From c66204b69ba201cf6611daae7e515442eed4b9cd Mon Sep 17 00:00:00 2001
From: llyyr <llyyr.public@gmail.com>
Date: Sat, 27 Dec 2025 02:19:15 +0530
Subject: [PATCH] screenshot: correctly detect hardware frame

This only ever worked by accident and relied on broken behavior that was
fixed by 9b1d47ece11da8625591bb5817bbad622485074b
---
 player/screenshot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/player/screenshot.c b/player/screenshot.c
index c733a2bdb1490..32d10a047db58 100644
--- a/player/screenshot.c
+++ b/player/screenshot.c
@@ -377,7 +377,7 @@ static struct mp_image *screenshot_get(struct MPContext *mpctx, int mode,
     }
 
     // vo_get_current_frame() can return a hardware frame, which we have to download first.
-    if (image && image->fmt.flags & MP_IMGFLAG_HWACCEL) {
+    if (image && IMGFMT_IS_HWACCEL(image->imgfmt)) {
         struct mp_image *nimage = mp_image_hw_download(image, NULL);
         talloc_free(image);
         if (!nimage)
-- 
2.51.0
